Lib/test/test_builtin.py

```diff
@@ -2130,14 +2130,11 @@ def test_warning_notimplemented(self):
         # be evaluated in a boolean context (virtually all such use cases
         # are a result of accidental misuse implementing rich comparison
         # operations in terms of one another).
-        # For the time being, it will continue to evaluate as a true value, but
-        # issue a deprecation warning (with the eventual intent to make it
-        # a TypeError).
-        self.assertWarns(DeprecationWarning, bool, NotImplemented)
-        with self.assertWarns(DeprecationWarning):
+        self.assertRaises(TypeError, bool, NotImplemented)
+        with self.assertRaises(TypeError):
```

Comment:
TypeError is raised when you just call a function with wrong number of arguments, for example:
```py
with self.assertRaises(TypeError):
    self.assertEqual(NotImplemented)
```
It is better to not use `assertTrue()` (which does not work here), but directly test the boolean value:
```py
if NotImplemented:
    pass
```

Reply:
I'll also change it to use `assertRaisesRegex` to test that we're getting the expected TypeError and not some other one.